pkg/dockerregistry/server/quotarestrictedblobstore.go

```diff
@@ -120,26 +161,103 @@ func (bw *quotaRestrictedBlobWriter) Commit(ctx context.Context, provisional dis
 	return can, err
 }
 
-// admitBlobWrite checks whether the blob does not exceed image quota, if set. Returns
+// admitBlobWrite checks whether the blob does not exceed image quota or limit ranges, if set. Returns
 // ErrAccessDenied error if the quota is exceeded.
-func admitBlobWrite(ctx context.Context, repo *repository) error {
-	rqs, err := repo.quotaClient.ResourceQuotas(repo.namespace).List(kapi.ListOptions{})
-	if err != nil {
-		if kerrors.IsForbidden(err) {
-			context.GetLogger(ctx).Warnf("Cannot list resourcequotas because of outdated cluster roles: %v", err)
-			return nil
-		}
-		context.GetLogger(ctx).Errorf("Failed to list resourcequotas: %v", err)
+func admitBlobWrite(ctx context.Context, repo *repository, size int64) error {
+	if err := admitLimitRanges(ctx, repo, size); err != nil {
+		return err
+	}
+
+	if err := admitQuotas(ctx, repo); err != nil {
 		return err
 	}
 
+	return nil
+}
+
+// admitLimitRanges checks the blob against any established limit ranges.
+func admitLimitRanges(ctx context.Context, repo *repository, size int64) error {
+	if size < 1 {
+		return nil
+	}
+
+	var (
+		lrs *kapi.LimitRangeList
+		err error
+	)
+
+	if projectLimitRanges != nil {
+		obj, exists, _ := projectLimitRanges.Get(repo.namespace)
+		if exists {
+			lrs = obj.(*kapi.LimitRangeList)
+		}
+	}
+	if lrs == nil {
+		context.GetLogger(ctx).Debugf("Listing limit ranges in namespace %s", repo.namespace)
+		lrs, err = repo.limitClient.LimitRanges(repo.namespace).List(kapi.ListOptions{})
+		if err != nil {
```

Comment:
> I think I'm misunderstanding what this is doing. Aren't errors when getting limitranges and resourcequotas supposed to be ignored.

@smarterclayton did you end up coming to the conclusion that we would detect the mode at startup?